### ISSUE TITLE
formatting problems with multiple tables and echo: fenced

### ISSUE THREAD

**Issue by richardjtelford** [quarto-dev/quarto-cli#2769]
### Bug description

When I use `echo: fenced` block option and have multiple tables in the block, the second table loses the sub-caption and the bold formatting of the header row. With `echo: true` everything is fine. 

Here is a minimal example

```` md
---
title: results
format: html
---
 
```{r}
#| echo: fenced
#| label: tbl-fenced
#| tbl-cap: "Two tables"
#| tbl-subcap: 
#|   - tab1
#|   - tab2

gt::gt(head(mtcars))
gt::gt(head(mtcars))
```

````

quarto --version  1.2.198
Ubuntu 18.04.6 LTS
RStudio 2022.07.1+554 "Spotted Wakerobin" Release (7872775ebddc40635780ca1ed238934c3345c5de, 2022-07-22) for Ubuntu Bionic

### Checklist

- [X] Please include a minimal, fully reproducible example in a single .qmd file? Please provide the whole file rather than the snippet you believe is causing the issue.
- [X] Please [format your issue](https://quarto.org/bug-reports.html#formatting-make-githubs-markdown-work-for-us) so it is easier for us to read the bug report.
- [X] Please document the RStudio IDE version you're running (if applicable), by providing the value displayed in the "About RStudio" main menu dialog?
- [X] Please document the operating system you're running. If on Linux, please provide the specific distribution.

**Comment by cscheid** [comment]
@cderv I think this might be a knitr issue? Check this out:

If we render with `echo: fenced`, only one cell-output-display div is created. The following .qmd:

````
```{r}
#| echo: fenced
#| label: tbl-fenced
#| tbl-cap: "Two tables"
#| tbl-subcap: 
#|   - tab1
#|   - tab2

gt::gt(head(mtcars, 1))
gt::gt(head(mtcars, 1))
```
````

produces this .md:

`````
::: {#tbl-fenced .cell tbl-cap='Two tables' tbl-subcap='["tab1","tab2"]'}

````{.cell-code}
```{{r}}
#| label: tbl-fenced
#| tbl-cap: "Two tables"
#| tbl-subcap:
#|   - tab1
#|   - tab2

gt::gt(head(mtcars, 1))
gt::gt(head(mtcars, 1))
```
````

::: {.cell-output-display}

```{=html}
<div id="ocowdyeuyg" style="overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
... gt output here
</div>
```

```{=html}
<div id="jmmgxvohco" style="overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
... gt output here
</div>
```

:::
:::
`````

But if we use, instead, this:

````
```{{r}}
#| label: tbl-fenced-2
#| tbl-cap: "Two tables"
#| tbl-subcap: 
#|   - tab1
#|   - tab2

gt::gt(head(mtcars, 1))
gt::gt(head(mtcars, 1))
```

```{r}
#| echo: false
#| label: tbl-fenced-2
#| tbl-cap: "Two tables"
#| tbl-subcap: 
#|   - tab1
#|   - tab2

gt::gt(head(mtcars, 1))
gt::gt(head(mtcars, 1))
```
````

(that is, emulating `echo: fenced` manually) then we get _two_ divs with `cell-output-display`:

`````
```{{r}}
#| label: tbl-fenced-2
#| tbl-cap: "Two tables"
#| tbl-subcap: 
#|   - tab1
#|   - tab2

gt::gt(head(mtcars, 1))
gt::gt(head(mtcars, 1))
```


::: {#tbl-fenced-2 .cell tbl-cap='Two tables' tbl-subcap='["tab1","tab2"]'}
::: {.cell-output-display}

```{=html}
<div id="zrbclxuqta" style="overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
...
</div>
```

:::

::: {.cell-output-display}

```{=html}
<div id="jmduetdahk" style="overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
...
</div>
```

:::
:::
`````

**Comment by cderv** [comment]
I don't think it is a **knitr** issue directly. `echo: fenced` is not a native **knitr** option and only available in Quarto. We have a special option hook that will change the default **knitr** behavior

https://github.com/quarto-dev/quarto-cli/blob/3db01ffdfee15159b8ef99429c9f24feafa02fb2/src/resources/rmd/hooks.R#L30-L48

So if you want to emulate it, you also need to set `fig.show: hold` and `results: hold` and this will explain the behavior

````markdown
---
title: results
format: html
keep-md: true
---
 
```{{r}}
#| label: tbl-fenced-2
#| tbl-cap: "Two tables"
#| tbl-subcap: 
#|   - tab1
#|   - tab2
#| fig-show: hold
#| results: hold

gt::gt(head(mtcars, 1))
gt::gt(head(mtcars, 1))
```

```{r}
#| echo: false
#| label: tbl-fenced-2
#| tbl-cap: "Two tables"
#| tbl-subcap: 
#|   - tab1
#|   - tab2
#| fig-show: hold
#| results: hold

gt::gt(head(mtcars, 1))
gt::gt(head(mtcars, 1))
```
````

You'll get the same - only one cell output display (because of `results: hold`) 
![image](https://user-images.githubusercontent.com/6791940/203749776-ec5b78b4-d330-4844-a386-38f4e284f68e.png)

We have `results: hold` by default for this feature otherwise we would get the code block with option twice 

![image](https://user-images.githubusercontent.com/6791940/203751384-74bfb843-0b97-4f17-ad35-582fa0affb59.png)

With `echo: fenced`, I believe we want to have the whole code chunk not spit and all results after the code chunk. Hence the `results: hold` option.

So we need to "fix" our `echo: fenced` feature to output what we need for our filters, or handle this special case in our filters for subcaption and all. 

What do you think ? Does it clarify or did I missed something else which make you think this is knitr issue ?

**Comment by cscheid** [comment]
> What do you think ? Does it clarify or did I missed something else which make you think this is knitr issue ?

Thank you, I think you're exactly right. I apologize for missing the additional knitr context. I thought it was a bug because I saw the conversion from fenced: true to fenced.echo: true. I missed the additional `hold` context.

Unless this is a very small, trivial fix, I think this is a sufficiently niche situation. I'm comfortable with leaving this bug open for a bit, and telling people that `echo: fenced` only works with a single output for now. Do you agree?

**Comment by cderv** [comment]
> Unless this is a very small, trivial fix, I think this is a sufficiently niche situation. I'm comfortable with leaving this bug open for a bit, and telling people that echo: fenced only works with a single output for now. Do you agree?

I don't see a trivial fix for now. I believe this is something to solve in Quarto where we need to align `echo: fenced` results with our filter that requires / expects one `::: {.cell-output-display}` per table or figure for cross reference. 

I need to see if we can tweak the hooks we have for **knitr** in Quarto to account for that. So for now I would say 

`echo: fenced` does not work with cross referencing well when multi figures or tables. Workaround is to split in two chunk: One unevaluated and one evaluated with no echo. 

Regarding niche situation that could be as I am surprised this was not found earlier.
